src/signalrclient/cancellation_token_source.h

```diff
@@ -7,57 +7,79 @@
 #include <assert.h>
 #include <condition_variable>
 #include <mutex>
+#include <vector>
+#include "signalrclient/cancellation_token.h"
 
 namespace signalr
 {
     class canceled_exception : public std::exception
     {
     };
 
-    class cancellation_token
+    class cancellation_token_source
     {
     public:
-        static const unsigned int timeout_infinite = 0xFFFFFFFF;
+        static const unsigned int timeout_infinite;
 
-        cancellation_token() noexcept
+        cancellation_token_source() noexcept
             : m_signaled(false)
         {
         }
 
-        cancellation_token(const cancellation_token&) = delete;
-        cancellation_token& operator=(const cancellation_token&) = delete;
+        cancellation_token_source(const cancellation_token_source&) = delete;
+        cancellation_token_source& operator=(const cancellation_token_source&) = delete;
 
-        ~cancellation_token()
+        ~cancellation_token_source()
         {
-            if (m_callback)
+            if (!m_callbacks.empty())
             {
-                m_callback();
+                for (auto& func : m_callbacks)
+                {
+                    try
+                    {
+                        func();
+                    }
+                    catch (...)
+                    {
+                        assert(false);
+                    }
+                }
```

Comment:
CTS's don't treat disposal as cancellation in C#, so I don't think we should treat destruction as cancellation here either. If the code using the CTS wants to always cancel before cleanup, it should do so explicitly.